### ISSUE TITLE
CHD v5 problem, audio stop

### ISSUE THREAD

**Issue by samaellain** [libretro/flycast#930]
Ready 2 Rumble Boxing Round 2 audio stop in middle of round and eventually core hang at end of round, retroarch menu still functional.
If i restart core in retroarch menu, game still hang.
No problem in GDI version.
Retroarch 1.8.9 core 0.1 4c8896d

save state seconds before music stop
[Ready 2 Rumble Boxing Round 2.zip](https://github.com/libretro/flycast/files/4881392/Ready.2.Rumble.Boxing.Round.2.zip)

**Comment by flyinghead** [comment]
I would suspect your CHD to be corrupted. I also have a CHDv5 of this game and I have no audio issue with it. This game doesn't play audio tracks from the disc so its format (chd, gdi, cue) shouldn't matter.

I tried your savestate and the audio didn't stop for me. It did hang at the end of the match though.

**Comment by samaellain** [comment]
Yes, game dont have music, the loop sound of the audience stop and in the end of the round hang.
in the matchs if the sound of the audience dont stop, the game dont hang and i can pass for the next match.
Is random the match the audience mute, but every time the sound stop, the round hang and dont go to next match.

**Comment by samaellain** [comment]
I use chdman 0.222 in the GDI, in case is important info.
In the GDi problem is not present.

**Comment by chrisfu** [comment]
I've been chatting to @rtissera in Discord about some other unrelated CHD v5 issues on Odroid Go Advance, and he's suggested integrating the current version of https://github.com/rtissera/libchdr to solve many issues with his original v5 implementation, particularly around ARM and 64-bit build targets.

Given that the static libchdr code within flycast is ~2 years old at this point, I'd imagine that this might be a reasonable step to take.

Certainly sounds as if dropping it in place in an experimental branch and seeing how well the integration tests turn out could be worth while, at least. :+1:
